CircuitBreaker doesn't work without `includes`

### Steps to Reproduce

1. Create two services, one communicating with the other using a declarative client with `@CircuitBreaker` annotated method (no `includes` or `excludes`).
2. Start only the main service and try to communicate with the other one using the client.

### Expected Behaviour

The connection is retried and after several attempts, the circuit is opened.


### Actual Behaviour

The `io.micronaut.http.client.exceptions.HttpClientException` exception is thrown, there is no retry. Logs:

```
06:44:46.241 [default-nioEventLoopGroup-1-2] ERROR i.m.h.s.netty.RoutingInBoundHandler - Unexpected error occurred: Connect error:Connection refused: localhost/127.0.0.1:8082
io.micronaut.http.client.exceptions.HttpClientException: Connect error:Connection refused: localhost/127.0.0.1:8082
        at io.micronaut.http.client.netty.DefaultHttpClient.lambda$null$31(DefaultHttpClient.java:1033)
        at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578)
        at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:571)
        at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:550)
        at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491)
        at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616)
        at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:609)
        at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:117)
        at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.fulfillConnectPromise(AbstractNioChannel.java:321)
        at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:337)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:707)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:655)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:581)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:8082
Caused by: java.net.ConnectException: Connection refused
        at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
        at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:779)
        at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:330)
        at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:707)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:655)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:581)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:834)
```

When the `io.micronaut.http.client.exceptions.HttpClientException` is added to `includes` of the `@CircuitBreaker`:
`@CircuitBreaker(includes = {io.micronaut.http.client.exceptions.HttpClientException.class})` it works as expected. Logs:

```
06:49:32.099 [default-nioEventLoopGroup-1-6] WARN  d.a.m.b.s.shared.RetryAttemptsLogger - Retry attempt recorded: io.micronaut.retry.event.RetryEvent[source=Flowable getRecommendations()]. Attempt: 1 of 3       
06:49:32.101 [default-nioEventLoopGroup-1-6] DEBUG i.m.r.i.DefaultRetryInterceptor - Retrying execution for method [Flowable getRecommendations()] after delay of 2000ms for exception: Connect error:Connection re
fused: localhost/127.0.0.1:8082                                                                        
io.micronaut.http.client.exceptions.HttpClientException: Connect error:Connection refused: localhost/127.0.0.1:8082
        at io.micronaut.http.client.netty.DefaultHttpClient.lambda$null$31(DefaultHttpClient.java:1033)            
        at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578)   
        at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:571)
        at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:550)                         
        at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491)                            
        at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616)                     

[...]

06:49:34.107 [default-nioEventLoopGroup-1-8] WARN  d.a.m.b.s.shared.RetryAttemptsLogger - Retry attempt recorded: io.micronaut.retry.event.RetryEvent[source=Flowable getRecommendations()]. Attempt: 2 of 3
06:49:34.107 [default-nioEventLoopGroup-1-8] DEBUG i.m.r.i.DefaultRetryInterceptor - Retrying execution for method [Flowable getRecommendations()] after delay of 3000ms for exception: Connect error:Connection re
fused: localhost/127.0.0.1:8082
io.micronaut.http.client.exceptions.HttpClientException: Connect error:Connection refused: localhost/127.0.0.1:8082
        at io.micronaut.http.client.netty.DefaultHttpClient.lambda$null$31(DefaultHttpClient.java:1033)
        at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578)
        at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:571)

[...]

06:49:41.129 [default-nioEventLoopGroup-1-12] DEBUG i.m.r.i.DefaultRetryInterceptor - Cannot retry anymore. Rethrowing original exception for method: Flowable getRecommendations()
06:49:41.129 [default-nioEventLoopGroup-1-12] DEBUG i.m.r.i.DefaultRetryInterceptor - Opening Circuit Breaker [Flowable getRecommendations()] due to error: Connect error:Connection refused: localhost/127.0.0.1:8
082
06:49:41.157 [default-nioEventLoopGroup-1-2] ERROR i.m.h.s.netty.RoutingInBoundHandler - Unexpected error occurred: Connect error:Connection refused: localhost/127.0.0.1:8082
io.micronaut.http.client.exceptions.HttpClientException: Connect error:Connection refused: localhost/127.0.0.1:8082
        at io.micronaut.http.client.netty.DefaultHttpClient.lambda$null$31(DefaultHttpClient.java:1033)
```

Neither the documentation nor the javadocs mention that `includes` is required.


### Environment Information

- **Operating System**: Ubuntu 18.04.5 LTS
- **Micronaut Version:** 2.3.1
- **JDK Version:** `OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.10+9, mixed mode)`

### Example Application

https://github.com/a-mroz/microservices-example/blob/master/shop/src/main/java/dev/amroz/microservices/bookshop/shop/infrastructure/RecommendationsClient.java

--- a/runtime/src/main/java/io/micronaut/retry/annotation/CircuitBreaker.java
+++ b/runtime/src/main/java/io/micronaut/retry/annotation/CircuitBreaker.java
@@ -17,10 +17,7 @@ package io.micronaut.retry.annotation;
 
 import static java.lang.annotation.RetentionPolicy.RUNTIME;
 
-import io.micronaut.aop.Around;
 import io.micronaut.context.annotation.AliasFor;
-import io.micronaut.context.annotation.Type;
-import io.micronaut.retry.intercept.DefaultRetryInterceptor;
 
 import javax.validation.constraints.Digits;
 import java.lang.annotation.Documented;

--- a/runtime/src/main/java/io/micronaut/retry/annotation/CircuitBreaker.java
+++ b/runtime/src/main/java/io/micronaut/retry/annotation/CircuitBreaker.java
@@ -38,8 +38,7 @@ import java.lang.annotation.Target;
 @Documented
 @Retention(RUNTIME)
 @Target({ElementType.METHOD, ElementType.TYPE, ElementType.ANNOTATION_TYPE})
-@Around
-@Type(DefaultRetryInterceptor.class)
+@Retryable
 public @interface CircuitBreaker {
 
     int MAX_RETRY_ATTEMPTS = 4;

--- a/runtime/src/test/groovy/io/micronaut/retry/intercept/CircuitBreakerSpec.groovy
+++ b/runtime/src/test/groovy/io/micronaut/retry/intercept/CircuitBreakerSpec.groovy
@@ -175,6 +175,18 @@ class CircuitBreakerSpec extends Specification{
         context.stop()
     }
 
+    void "test circuit breaker empty config"() {
+        given:
+        ApplicationContext context = ApplicationContext.run()
+        EmptyConfigService service = context.getBean(EmptyConfigService)
+
+        when:"A method is annotated retry"
+        int result = service.getCount()
+
+        then:"It executes until successful"
+        result == 2
+    }
+
     @Singleton
     static class MyRetryListener implements RetryEventListener {
 
@@ -237,5 +249,21 @@ class CircuitBreakerSpec extends Specification{
             })
         }
 
+    }
+
+    @Singleton
+    @CircuitBreaker
+    static class EmptyConfigService {
+        int countValue = 0
+        int countRx = 0
+        int countThreshold = 2
+
+        int getCount() {
+            countValue++
+            if(countValue < countThreshold) {
+                throw new IllegalStateException("Bad count")
+            }
+            return countValue
+        }
     }
 }

